cli/src/main/kotlin/io/provenance/kafka/cli/Main.kt

```diff
@@ -54,26 +54,58 @@ fun main(args: Array<String>) {
 
     logger("org.apache.kafka").level = Level.WARN
 
-    val incoming = kafkaChannel<String, String>(commonProps, setOf(source))
-        .receiveAsFlow()
-        .buffer()
+    val incoming = kafkaConsumerChannel<String, String>(commonProps, setOf(source))
+    val producer = kafkaProducerChannel<String, String>(commonProps)
 
     runBlocking {
+
+        //
+        // Using select
+        //
+
         launch(Dispatchers.IO) {
-            incoming.onEachToTopic(commonProps) { ProducerRecord(dest, it.key(), it.value()) }
-                .buffer()
-                .acking {
-                    logger("main").info("acking record: ${it.key()} // ${it.value()}")
+            val ticker = ticker(5000)
```

Comment:
what is the reason for applying a ticker? curious if we need to add a ticker mode along with it so we are kind of aware of the purpose?

Reply:
It's just an example to periodically produce messages to the source topic so we have something to consume.

Reply:
Added clarifying comments around the ticker.